### PR TITLE
117 entries template

### LINKED ISSUE #117: Entries Template
Per our feedback from our surveys, they wanted us to, in some cases, frame the journal entry so that the user can have some direction for writing their entry.

### PR DESCRIPTION
**Pull request by S1robe**
# Description
This PR adds templates to the entries which are a direct result from survey feedback. Templates are a way for users to guide themselves during a journaling session. This PR also updates the plans to the Design doc

Fixes #117 #115 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Modified the entries_test.dart to open the dialogue to generate a template for the journal entry.

**Test Configuration**:
    • Platform android-34, build-tools 34.0.0
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

image
image
image

### CODE REVIEW

**Review by ian5924 — APPROVED** [review]
Good job.
The template questions look good.

File: app/integration_test/calendar_test.dart [hunk 1]
```diff
@@ -112,7 +112,7 @@ void main() {
 			JournalEntry(
 				title: "Title!",
 				entryText: "Journal!",
-				date: today,
+				dateOverride: today,
```
Comment (Benek6h) [inline app/integration_test/calendar_test.dart:115]: Nice catch. Thanks for fixing the bugs with the date.

File: app/lib/pages/entries.dart [hunk 2]
```diff
@@ -113,21 +111,18 @@ class _EntryPanelPageState extends State<EntryPanelPage> {
...
+                          child: TextFormField(
+                            key: const Key('Filter_By_TextForm'),
+                            textAlign: TextAlign.center,
+                            controller: searchBarInput,
+                            onChanged: updateFilteredList,
+                            decoration: const InputDecoration(
+                                border: UnderlineInputBorder(),
+                                labelText: 'Enter a journal title',
+                                fillColor: Colors.transparent),
+                          )),
```
Comment (Benek6h) [inline app/lib/pages/entries.dart:125]: Thanks for expanding the filtering to plans!

File: app/lib/provider/entry.dart [hunk 3]
```diff
@@ -58,6 +58,96 @@ Map<String, Color> emotionList = {
...
+  while(indexes.length != num){indexes.add(Random().nextInt(length));}
+  return indexes.map((e) => questions[e]).join("\n\n\n"); // pull those questions and join them.
+}
+
+(String, String) getTemplateRandom() {
+  String cat, body = "";
+  cat = templateSet.keys.toList()[Random().nextInt(templateSet.length)];
+  body = templateSet[cat]![Random().nextInt(templateSet[cat]!.length)];
+  return (cat, body);
+}
```
Comment (Benek6h) [inline app/lib/provider/entry.dart:148]: Nice job with the template, keeping the code short and precise.

File: app/lib/uiwidgets/decorations.dart [hunk 4]
```diff
@@ -453,113 +465,84 @@ class _DisplayCardState extends State<DisplayCard> {
...
+                                      Theme.of(context).primaryColor,
+                                  decorationThicknessFactor: 1,
+                                  decorationThicknessDelta: 0.0,
+                                )
+                              // Otherwise no text style changes
+                              : DefaultTextStyle.of(context).style.apply(
+                                  fontSizeFactor: 1.3,
+                                  fontWeightDelta: 1,
+                                ),
+                        ),
```
Comment (Benek6h) [inline app/lib/uiwidgets/decorations.dart:496]: Thank you for reimplementing the front end for the display card.

**Review by Benek6h — APPROVED** [review]
This is great work! Good job, Garrett!

**Pull request merged**